apps/emqx_modules/src/emqx_modules_conf.erl

```diff
@@ -135,6 +160,18 @@ post_config_update(
     case emqx_topic_metrics:deregister(Topic) of
         ok -> ok;
         {error, Reason} -> {error, Reason}
+    end;
+post_config_update(_, _UpdateReq, NewConfig, OldConfig, _AppEnvs) ->
+    #{
+        removed := Removed,
+        added := Added
+    } = emqx_utils:diff_lists(NewConfig, OldConfig, fun(#{topic := T}) -> T end),
+    Deregistered = [emqx_topic_metrics:deregister(T) || #{topic := T} <- Removed],
+    Registered = [emqx_topic_metrics:register(T) || #{topic := T} <- Added],
+    Errs = [Res || Res <- Registered ++ Deregistered, Res =/= ok],
+    case Errs of
+        [] -> ok;
+        _ -> {error, Errs}
     end.
```

Comment:
Nice catch, I can't believe I don't implment config_handler's callback for topic_metrics before.
Nit:
deregister will return ok or topic_not_found, we should mark topic_not_found as deregister success.
register will return ok or already_existed,  maybe already_existed is also register success.

Reply:
Thanks, @zhongwencool, fixed.